webui/test/integration/utils/mockServer.ts

```diff
@@ -0,0 +1,180 @@
+        resolve();
+      }
+    });
+  }
+
+  private routeRequest(req: any, res: any) {
+    const requestPath = req.url?.split('?')[0];
+
+    if (req.method === 'OPTIONS') {
+      return this.handleCorsPreflightRequest(res);
+    }
+
+    if (requestPath && this.apiMocks.has(requestPath)) {
+      return this.handleMockRequest(req, res);
+    }
+
+    if (req.url?.startsWith('/api/')) {
+      return this.handleUnmockedApiRequest(res);
+    }
+
+    return this.handleStaticFileRequest(req, res);
+  }
+
+  private handleMockRequest(req: any, res: any) {
+    try {
+      const requestPath = req.url?.split('?')[0];
+      const mockResponse = this.apiMocks.get(requestPath);
+
+      this.validateMockResponse(mockResponse, requestPath);
+      
+      const headers = this.buildResponseHeaders(mockResponse);
+      
+      res.writeHead(mockResponse.status, headers);
+      res.end(mockResponse.body);
+    } catch (error) {
+      this.sendErrorResponse(res);
+    }
+  }
+
+  private validateMockResponse(mockResponse: any, requestPath: string) {
+    if (typeof mockResponse.status !== 'number') {
+      throw new Error(`Mock for ${requestPath} missing required 'status' field`);
+    }
+  }
+
+  private buildResponseHeaders(mockResponse: any) {
+    const headers = {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+      'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+      ...mockResponse.headers
+    };
+
+    if (mockResponse.contentType) {
+      headers['Content-Type'] = mockResponse.contentType;
+    } else if (mockResponse.body) {
+      headers['Content-Type'] = 'application/json';
+    }
+
+    return headers;
+  }
+
+  private sendErrorResponse(res: any) {
+    res.writeHead(500, {
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    res.end(JSON.stringify({ message: 'Internal server error' }));
+  }
+
+  private handleUnmockedApiRequest(res: any) {
+    res.writeHead(404, {
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    res.end(JSON.stringify({ message: 'API endpoint not mocked' }));
+  }
+
+  private handleCorsPreflightRequest(res: any) {
+    res.writeHead(200, {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+      'Access-Control-Allow-Headers': 'Content-Type, Authorization'
+    });
+    res.end();
+  }
+
+  private handleStaticFileRequest(req: any, res: any) {
+    try {
+      if (!req.url) {
+        this.serveIndexHtml(res);
+        return;
+      }
+
+      let filePath = req.url === '/' ? '/index.html' : req.url;
+      if (filePath?.includes('?')) {
+        filePath = filePath.split('?')[0];
+      }
+
+      if (!filePath || filePath === '/') {
+        filePath = '/index.html';
+      }
+      
+      const fullPath = join(process.cwd(), 'dist', filePath);
```

Comment:
1. import and use path's normalize too
2. cwd is risky suggest to use the source location for relative location and not the location I started nodejs.